sdk/core/Azure.Core/samples/Configuration.md

```diff
@@ -103,6 +103,55 @@ SecretClientOptions options = new SecretClientOptions()
 };
 ```
 
+Another customization that is possible is to easily configure what exceptions or status codes 
+are retried. By default, this is configured in the `ResponseClassifier` type, but it can be 
+overridden by setting the `ResponseClassifier` property on the `RetryPolicyOptions` when 
+constructing a `RetryPolicy`. In the example below, we create a custom response classifier that 
+we use in our custom `RetryPolicy`.
+```C# Snippet:CustomResponseClassifier
+public class CustomResponseClassifier : ResponseClassifier
+{
+    public override bool IsRetriableResponse(HttpMessage message)
+    {
+        switch (message.Response.Status)
+        {
+            case 404:
+                return true;
+            default:
+                return base.IsRetriableResponse(message);
```

Comment:
Question: the note on line 204 re: when library-specific response classifiers will and won't be respected makes me wonder if we should consider shifting the design so that it always calls through to the base if the caller doesn't handle every case.  This would make it less error prone if someone forgot to call the base, but it may not be worth the complexity and/or may not be the desired behavior.

Reply:
This was based on my port of test infrastructure in #37568.  I wanted to include not only additional response codes but also custom error logic - basically, I wanted the full control of a classifier and ability to control the logic without having to write a custom retry policy just to inject one.

Reply:
Yeah, I don't think we do this in general as it relies on a more complex surface. It is the overrider's responsibility to handle the different statuses as they see fit.